measureme/src/profiling_data.rs

```diff
@@ -0,0 +1,73 @@
+use crate::event::Event;
+use crate::{ProfilerFiles, RawEvent, StringTable};
+use std::fs;
+use std::mem;
+use std::path::Path;
+use std::time::{Duration, SystemTime};
+
+pub struct ProfilingData {
+    event_data: Vec<u8>,
+    string_table: StringTable,
+}
+
+impl ProfilingData {
+    pub fn new(path_stem: &Path) -> ProfilingData {
+        let paths = ProfilerFiles::new(path_stem);
+
+        let string_data = fs::read(paths.string_data_file).expect("couldn't read string_data file");
+        let index_data = fs::read(paths.string_index_file).expect("couldn't read string_index file");
+        let event_data = fs::read(paths.events_file).expect("couldn't read events file");
+
+        let string_table = StringTable::new(string_data, index_data);
+
+        ProfilingData {
+            string_table,
+            event_data,
+        }
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = Event<'_>> {
+        ProfilerEventIterator {
+            data: self,
+            curr_event_idx: 0,
+        }
+    }
+}
+
+struct ProfilerEventIterator<'a> {
+    data: &'a ProfilingData,
+    curr_event_idx: usize,
+}
+
+impl<'a> Iterator for ProfilerEventIterator<'a> {
+    type Item = Event<'a>;
+
+    fn next(&mut self) -> Option<Event<'a>> {
+        let raw_idx = self.curr_event_idx * mem::size_of::<RawEvent>();
+        let raw_idx_end = raw_idx + mem::size_of::<RawEvent>();
+        if raw_idx_end > self.data.event_data.len() {
+            return None;
+        }
+
+        self.curr_event_idx += 1;
+
+        let raw_event_bytes = &self.data.event_data[raw_idx..raw_idx_end];
+
+        let raw_event = unsafe {
+            &*(raw_event_bytes.as_ptr()  as *const RawEvent)
+        };
+
+        let string_table = &self.data.string_table;
+
+        let mut timestamp = SystemTime::UNIX_EPOCH;
```

Comment:
The way `Profiler::record_event()` works at the moment, the timestamp contains the number of nanoseconds since `Profiler::start_time`, not since `SystemTime::UNIX_EPOCH`.

Reply:
I think the reason I changed this was because there was no way to reconstruct the `Instant` values since the only `Instant` constructor is `Instant::now()`. It'd also be nice to have `SystemTime` instances here so that we can do things like have an absolute timeline of events even across multiple, possibly parallel `rustc` processes.